### PR TITLE
Add dependencies section to maintain citation references.

### PR DESCRIPTION
**Pull request by rmcilroy**
Addressing comment on #59

### CODE REVIEW

**Comment by rmcilroy** [comment]
Ilya, I made this change as suggested, however the links to the HTML spec don't work properly. They link to the title page of the multi-page version, so the fragment identifiers don't point to anything on that page (they are in one of the sub-pages). Any ideas?

**Comment by igrigorik** [comment]
Paging @marcoscaceres @siusin for x-ref guidance.. :-)

File: index.html [hunk 1]
```diff
@@ -80,6 +80,33 @@
...
     <p>The Web Performance Working Group maintains a <a href="https://github.com/w3c/web-platform-tests/tree/master/html/webappapis/idle-callbacks">test suite</a> and an <a href="https://w3c.github.io/test-results/requestidlecallback/">implementation report</a> for the specification. Vendors interested in implementing this document SHOULD join the mailing lists below and take part in the discussions.</p>
   </section>
+  <section>
+    <h2>Dependencies</h2>
+    <p>The terms <dfn data-cite="html51#sec-browsing-contexts">browsing context
+    </dfn>, <dfn data-cite="html51#event-loop">event loop</dfn>,
+    <dfn data-cite="html51#event-loops-processing-model">event loop processing
+    model</dfn>, <dfn data-cite="html51#spinning-the-event-loop">spin the event
+    loop</dfn>, <dfn data-cite="html51#fully-active">fully active</dfn>,
+    <dfn data-cite="html51#tasks">tasks</dfn>,
```
Comment (marcoscaceres) [inline index.html]: Please cite "html" instead of "html51"

File: index.html [hunk 2]
```diff
@@ -89,9 +116,10 @@
     templating and pre-rendering of content likely to become visible in the
     near future. These tasks must share the event loop with other time-critical
     operations, such as reacting to input and performing script-based
-    animations using `requestAnimationFrame` ([[ANIMATION-TIMING]]). These
+    animations using <code><a>requestAnimationFrame</a></code>. These
```
Comment (marcoscaceres) [inline index.html]: remove `<code>`

Comment (marcoscaceres) [reply index.html]: If you need `<code>` anywhere for a method or attribute, it means that the method or attribute is not correctly defined. 

To correctly define an attribute or method from another spec, you do it like this:

```HTML
<code><dfn data-cite="!html#dom-window-requestanimationframe">requestAnimationFrame()</dfn></code> is defined in [[!html]].
```

File: index.html [hunk 3]
```diff
@@ -89,9 +116,10 @@
...
     background tasks are typically performed by scheduling a callback using
-    `[setTimeout]` and running the background task during that callback.</p>
+    <code><a>setTimeout</a></code> and running the background task during that
```
Comment (marcoscaceres) [inline index.html]: As above.

File: index.html [hunk 4]
```diff
@@ -313,7 +341,7 @@
 
 callback IdleRequestCallback = void (IdleDeadline deadline);
 </pre>
-    <p>Each `[Window]` has:</p>
+    <p>Each <code><a>Window</a></code> has:</p>
```
Comment (marcoscaceres) [inline index.html]: Same here... and I guess the rest :)

File: index.html [hunk 5]
```diff
@@ -80,6 +80,33 @@
...
+    <dfn data-cite="html51#event-loops-processing-model">event loop processing
+    model</dfn>, <dfn data-cite="html51#spinning-the-event-loop">spin the event
+    loop</dfn>, <dfn data-cite="html51#fully-active">fully active</dfn>,
+    <dfn data-cite="html51#tasks">tasks</dfn>,
+    <dfn data-cite="html51#task-source">task source</dfn>,
+    <dfn data-cite="html51#task-queues">task queues</dfn>,
+    <dfn data-cite="html51#queuing">queue a task</dfn>,
+    <dfn data-cite="html51#microtask-queue">microtask queue</dfn>,
+    <dfn data-cite="html51#list-of-active-timers">list of active timers</dfn>,
+    <dfn data-cite="html51#dom-windowtimers-settimeout">setTimeout</dfn>,
```
Comment (marcoscaceres) [inline index.html]: wrap these ones in `<code>` and see below... then you don't need `<code>` anywhere else and you the spec will be auto-linked correctly.

Comment (rmcilroy) [reply index.html]: done

File: index.html (hunk 4 above)
Comment (rmcilroy) [reply index.html]: done

File: index.html (hunk 3 above)
Comment (rmcilroy) [reply index.html]: done

File: index.html (hunk 2 above)
Comment (rmcilroy) [reply index.html]: Done, thanks for the hint.

File: index.html (hunk 1 above)
Comment (rmcilroy) [reply index.html]: Done

**Comment by rmcilroy** [comment]
Comments addressed, thanks for the hints, that seemed to work.

**Comment by igrigorik** [comment]
Hmm, I'm seeing 4 warnings on this end. Preview: https://cdn.rawgit.com/rmcilroy/requestidlecallback/dependencies/index.html

image

Are you seeing same warnings?

**Comment by rmcilroy** [comment]
I don't see those errors. This PR shouldn't have changed those links either. Did you try with the non-cdn version of rawgit?

**Comment by igrigorik** [comment]
Yep, cmd-F5 and I still see them. 

Ditto if I view the blob URL linked in this PR:
https://github.com/rmcilroy/requestidlecallback/blob/ad7f456c52831d49be6a22bcc09a4dffc0546132/index.html

**Comment by marcoscaceres** [comment]
I'm also not seeing any errors on:
https://cdn.rawgit.com/rmcilroy/requestidlecallback/dependencies/index.html

**Comment by igrigorik** [comment]
What the.. Something really odd is happening here then. I'm using incognito + CDT disable cache and I keep seeing the same error. Here's the trace HAR trace: https://gist.github.com/igrigorik/ffdb787e5ad16e456b94d89bf7a30fbb#file-ric-har-json-L166

Screenshot from devtools:
image

igrigorik { ~ } > curl -s https://www.w3.org/Tools/respec/respec-w3c-common | md5
ffce5b58bff3c09934147abca2d7734a

**Comment by marcoscaceres** [comment]
Strange. Maybe let's just get this merged? If the warnings are still there, I can take over and do a markup cleanup and squash any remaining issues.

**Comment by marcoscaceres** [comment]
Seems good! No warnings on the gh-pages branch (https://w3c.github.io/requestidlecallback/). 

🍾  to celebrate, I just  [used `requestIdleCallback()` in ReSpec](https://github.com/w3c/respec/pull/1364/files#diff-54952926011bc1e0b753e3e94681c691R18) to schedule some low priority stuff 😀

**Comment by igrigorik** [comment]
Bizarro.. @marcoscaceres I'm still seeing same errors =\

image

**Comment by marcoscaceres** [comment]
So super weird. I've tried in multiple browsers, cleared caches, etc. Here is a screenshot from Chrome Canary: 

screenshot 2017-08-15 13 52 15

@rmcilroy or anyone else, can you confirm that it's working?

**Comment by marcoscaceres** [comment]
Here is Safari - showing cache disabled (and cache cleared) and also no warnings (forced lots reloads, just to be sure):

screenshot 2017-08-15 13 56 30

**Comment by marcoscaceres** [comment]
Also checked on a different computer in Edge. Seems happy there too. 

Seems the interwebs is trolling you, @igrigorik.

**Comment by igrigorik** [comment]
/me facepalm.. apparently, it's Ghostery! As soon as I disable it, everything is a-OK. Apologies for the noise.

**Comment by siusin** [comment]
Thanks @rmcilroy !

@igrigorik the latest commit looks good to me, could you try this URL?
[https://rawgit.com/w3c/requestidlecallback/ad7f456/index.html](https://rawgit.com/w3c/requestidlecallback/ad7f456/index.html)

**Comment by rmcilroy** [comment]
Thanks everyone for investigating!

> :champagne: to celebrate, I just used requestIdleCallback() in ReSpec to schedule some low priority stuff :grinning:

Awesome stuff :+1:

**Pull request merged**